yt/frontends/amrvac/tests/test_outputs_pytest.py

```diff
@@ -0,0 +1,252 @@
+    jet_cylindrical_25D,
+    riemann_cartesian_175D,
+    blastwave_cartesian_3D,
+    blastwave_polar_2D,
+    blastwave_cylindrical_3D,
+    rmi_cartesian_dust_2D,
+]
+a_list = [0, 1, 2]
+w_list = [None, "density"]
+dso_list = [None, ("sphere", ("max", (0.1, "unitary")))]
+
+
+# Tests for units: verify that overriding certain units yields the correct derived units.
+# The following are correct normalisations based on length, numberdensity and temperature
+length_unit = (1e9, "cm")
+numberdensity_unit = (1e9, "cm**-3")
+temperature_unit = (1e6, "K")
+density_unit = (2.341670657200000e-15, "g*cm**-3")
+mass_unit = (2.341670657200000e12, "g")
+velocity_unit = (1.164508387441102e07, "cm*s**-1")
+pressure_unit = (3.175492240000000e-01, "dyn*cm**-2")
+time_unit = (8.587314705370271e01, "s")
+magnetic_unit = (1.997608879907716, "gauss")
+
+
+def _get_fields_to_check(fname):
+    # This function is called during test collection. If this frontend
+    # is not being run, and therefore the data isn't present, this try
+    # except block prevents pytest from failing needlessly
+    try:
+        ds = utils.data_dir_load(fname)
+        fields = [("gas", "density"), ("gas", "velocity_magnitude")]
+        field_ids = ["density", "velocity_magnitude"]
+        raw_fields_labels = [fname for ftype, fname in ds.field_list]
+        if "b1" in raw_fields_labels:
+            fields.append(("gas", "magnetic_energy_density"))
+            field_ids.append("magnetic_energy_density")
+        if "e" in raw_fields_labels:
+            fields.append(("gas", "energy_density"))
+            field_ids.append("energy_density")
+        if "rhod1" in raw_fields_labels:
+            fields.append(("gas", "total_dust_density"))
+            field_ids.append("total_dust_density")
+            # note : not hitting dust velocity fields
+        # return [fields, field_ids]
+        return [ds, fields]
+    except YTOutputNotIdentified:
+        return [[pytest.param(None, marks=pytest.mark.skip),], ["Data not found",]]
+
+
+def get_pairs():
+    f_list = [_get_fields_to_check(fname) for fname in ds_list]
+    pairs = [(i[0], f) for i in f_list for f in i[1]]
+    return pairs
+
+
+def _assert_normalisations_equal(ds):
+    assert_allclose_units(ds.length_unit, YTQuantity(*length_unit))
+    assert_allclose_units(ds.numberdensity_unit, YTQuantity(*numberdensity_unit))
+    assert_allclose_units(ds.temperature_unit, YTQuantity(*temperature_unit))
+    assert_allclose_units(ds.density_unit, YTQuantity(*density_unit))
+    assert_allclose_units(ds.mass_unit, YTQuantity(*mass_unit))
+    assert_allclose_units(ds.velocity_unit, YTQuantity(*velocity_unit))
+    assert_allclose_units(ds.pressure_unit, YTQuantity(*pressure_unit))
+    assert_allclose_units(ds.time_unit, YTQuantity(*time_unit))
+    assert_allclose_units(ds.magnetic_unit, YTQuantity(*magnetic_unit))
+
+
+@pytest.mark.answer_test
+@pytest.mark.usefixtures("answer_file", "answer_compare")
+class TestAMRVac:
```

Comment:
not crucial but no one in this code community would use this case : this name is an acronym, it's not just the "amr" part :)
(if you're curious, VAC means "Versatile Advection Code" and is the name for AMRVAC's ancestor. The full name of the current code is actually MPI-AMRVAC but it doesn't exactly roll on the tongue so everyone uses "AMRVAC")

Reply:
Gotcha, thanks! I'll change it!